corehq/apps/domain/decorators.py

```diff
@@ -249,13 +247,29 @@ def _inner(request, *args, **kwargs):
     return decorator
 
 
+def two_factor_exempt(view_func):
+    """
+    Marks a view function as being exempt from two factor authentication.
+    """
+    # We could just do view_func.two_factor_exempt = True, but decorators
+    # are nicer if they don't have side-effects, so we return a new
+    # function.
+    def wrapped_view(*args, **kwargs):
+        return view_func(*args, **kwargs)
+    wrapped_view.two_factor_exempt = True
+    return wraps(view_func, assigned=available_attrs(view_func))(wrapped_view)
```

Comment:
Ah, is this the distinction?  (Checking my understanding)

In the common usage, they should behave identically:
```python
@two_factor_exempt
def view_func(request):
    return "hi"
```
But the simpler implementation mutates the function, so if you needed access to the wrapped and unwrapped versions:
```python
def view_func(request):
    return "hi"

two_factor_exempt_view_func = two_factor_exempt(view_func)
```
Then both `view_func` and `two_factor_exempt_view_func` would both (surprisingly) have the attr set?  That's quite a gotcha.